### PR TITLE
[eligibility] eligibility confirmation and case deletion

### PR DESCRIPTION
**Pull request by me-liu**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

- Ineligible screen styling + routing.
- Cases are deleted from user view if the user claims to be ineligible.

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

- Add new cases and make sure they show up properly.
- Decline eligibility and make sure the case is deleted from the `/AllCases` view (make sure that it deletes from the supabase 'status' table as well)

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

- Consider moving case related queries into the `CaseContext`.
- Standardize error handling for case queries and logic.

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[Original Case Context Implementation](https://github.com/calblueprint/impact-fund/pull/36)
[Eligibility Requirements Screen](https://github.com/calblueprint/impact-fund/pull/67)

CC: @ronniebeggs

### CODE REVIEW

File: src/app/(BottomTabNavigation)/AllCases/EligibilityForm/[caseUid].tsx [hunk 1]
```diff
@@ -14,7 +14,15 @@ export default function EligibilityForm() {
...
+        router.push({
+          pathname: 'AllCases/EligibilityForm/ConfirmEligibility',
+          params: { caseUid },
+        });
+      } else {
+        await updateCaseStatus(caseUid, Eligibility.ELIGIBLE);
+        router.back();
+      }
     }
   };
```
Comment (ronniebeggs) [inline src/app/(BottomTabNavigation)/AllCases/EligibilityForm/[caseUid].tsx]: Can we rewrite this function so it merely routes the users to the `ConfirmEligibility` screen. There are several `updateCaseStatus` calls being made here that I think we only make once they actually confirm their ineligibility.

Comment (ronniebeggs) [reply src/app/(BottomTabNavigation)/AllCases/EligibilityForm/[caseUid].tsx]: Only eligible users will be able to view this screen anyways.

File: src/app/(BottomTabNavigation)/AllCases/EligibilityForm/ConfirmEligibility/index.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          </Text>
+        </View>
+        <View style={styles.infoContainer}>
+          <LittlePerson style={{ marginRight: 17, marginTop: 12 }} />
+
+          <Text style={styles.infoText}>
+            You can still view and share case details, but won’t be able to
+            change your eligibility status or file a claim.{' '}
+          </Text>
+        </View>
```
Comment (ronniebeggs) [inline src/app/(BottomTabNavigation)/AllCases/EligibilityForm/ConfirmEligibility/index.tsx]: For these Icon-Text containers, it'd probably be easiest to style by placing a gap between them, and otherwise not setting their margins with the container. Based off the figma, it seems that the top of these icons should align with the top of the accompanying text.

File: src/app/(BottomTabNavigation)/AllCases/EligibilityForm/ConfirmEligibility/index.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+          style={styles.buttonBlack}
+          onPress={() => updateActivity(false)}
+        >
+          <View style={styles.center}>
+            <Check style={{ marginRight: 10 }} />
+            <Text style={styles.buttonText}>Yes, I do</Text>
+          </View>
+        </TouchableOpacity>
+      </View>
+    </View>
```
Comment (ronniebeggs) [inline src/app/(BottomTabNavigation)/AllCases/EligibilityForm/ConfirmEligibility/index.tsx]: There seems to be some whitespace below this container when the button container should directly touch the bottom tab bar. The spacing between the bar and the button should come from inner padding instead.

File: src/app/(BottomTabNavigation)/AllCases/index.tsx [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
 import CaseCard from '../../../Components/CaseCard/CaseCard';
 import { CaseContext } from '../../../context/CaseContext';
-import supabase from '../../../supabase/createClient';
 
 function CasesScreen() {
-  const { allCases, loading } = useContext(CaseContext);
-
+  //const { allCases } = useContext(CaseContext);
+  const { activeCases, loading } = useContext(CaseContext);
+  const { inactiveCases } = useContext(CaseContext);
```
Comment (ronniebeggs) [inline src/app/(BottomTabNavigation)/AllCases/index.tsx]: Right now, inactive cases don't immediately show up on the `AllCases` screen unless the app is otherwise reloaded. I'm not sure the best way to fix it, but let's try adding a navigation listener that requires the cases when the page is visited. 
Screen Shot 2024-03-20 at 5 51 18 PM

File: src/app/(BottomTabNavigation)/AllCases/EligibilityForm/ConfirmEligibility/index.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        </TouchableOpacity>
+        <TouchableOpacity
+          style={styles.buttonBlack}
+          onPress={() => updateActivity(false)}
+        >
+          <View style={styles.center}>
+            <Check style={{ marginRight: 10 }} />
+            <Text style={styles.buttonText}>Yes, I do</Text>
+          </View>
+        </TouchableOpacity>
```
Comment (ronniebeggs) [inline src/app/(BottomTabNavigation)/AllCases/EligibilityForm/ConfirmEligibility/index.tsx]: I realize this is not reflected on Figma, but can we make these buttons side by side for consistency with other screens?

**Review by ronniebeggs — COMMENTED** [review]
Coming along nicely so far. The styling in this PR looks much better! Just a few changes we should address during worksession.

File: src/app/(BottomTabNavigation)/AllCases/EligibilityForm/ConfirmEligibility/index.tsx (hunk 3 above)
Comment (me-liu) [reply src/app/(BottomTabNavigation)/AllCases/EligibilityForm/ConfirmEligibility/index.tsx]: Not sure if my fix is what you meant, but there isn't any extraneous padding on the bottom
